File: back-end/DAL_01/BOL_02/OrderModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace BOL_02
{
    public class OrderModel
    {
        public int OrderId { get; set; }
        public int UserID { get; set; }
        public int CarID { get; set; }

        [Required, DataType(DataType.DateTime)]
        public DateTime RentedDate { get; set; }

        [Required, DataType(DataType.DateTime)]
        public DateTime ReturnDate { get; set; }

        [DataType(DataType.DateTime)]
        public DateTime ActualReturnDate { get; set; }
    }
}

File: front-end/app/shared/models/order.model.ts
export class Order {
    UserID: number;
    CarID: number;
    RentedDate: Date;
    ReturnDate: Date;
    ActualReturnDate ?: Date;
    OrderId: number;
    constructor(UserID?: number, CarID?: number, rentedCar?: Date, returnCar?: Date , OrderId?: number) {
        this.UserID = UserID;
            this.CarID = CarID;
           this.RentedDate = rentedCar;
           this.ReturnDate = returnCar;
           this.OrderId = OrderId;
    }

    }
